src/main/java/com/iexec/worker/executor/TaskManagerService.java

```diff
@@ -451,7 +451,7 @@ boolean abort(String chainTaskId) {
         log.info("Stopped task containers [chainTaskId:{}]", chainTaskId);
         subscriptionService.unsubscribeFromTopic(chainTaskId);
         purgeService.purgeAllServices(chainTaskId);
-        boolean isSuccess = resultService.removeResult(chainTaskId);
+        boolean isSuccess = resultService.purgeTask(chainTaskId);
```

Comment:
If I understand correctly, as `ResultService` implements `Purgeable`, this method should be already called in the previous `purgeService.purgeAllServices(chainTaskId)` call.

I see an issue in the current implementation logic.
Why would the purge of `ResultService` be the most important one to determine if a task has been correctly aborted ?
What happens if the purge mechanism of any other purgeable service fails ?

Shouldn't we collect the status of all purgeTask calls in purgeService and return their aggregation in a single boolean value ?

Reply:
You're absolutely right! I was not as attentive as I should have been. Here's a fix: https://github.com/iExecBlockchainComputing/iexec-worker/pull/476/commits/d83419752aefe193e2fda1993164f957565227b6